main.go

```diff
@@ -227,35 +90,20 @@ func main() {
 	handler.SetFlags(flagSet)
 
 	// initialize the handler
-	err := handler.Initialize()
-
-	if err != nil {
+	if err := handler.Initialize(); err != nil {
 		log.Error(err.Error())
 		os.Exit(1)
 	}
 
 	// Parse all command-line flags
-	err = flagSet.Parse(os.Args[2:])
-
-	if err != nil {
+	if err := flagSet.Parse(os.Args[2:]); err != nil {
 		log.Error(err.Error())
 		os.Exit(1)
 	}
 
 	// Check that required flags/envs were set
-	if config.GetString(Primary.EnvVar) == "" {
-		log.Error("Must specify a primary PostgreSQL instance.")
-		log.Errorf("This value can be provided by either the '--%s' flag or the '%s' environment variable",
-			Primary.Name, Primary.EnvVar)
-		os.Exit(1)
-	}
-
-	if config.GetString(Replica.EnvVar) == "" {
-		log.Error("Must specify a replica PostgreSQL instance for failover.")
-		log.Errorf("This value can be provided by either the '--%s' flag or the '%s' environment variable",
-			Replica.Name, Replica.EnvVar)
-		os.Exit(1)
-	}
+	findConfigOrFail(Primary, "Must specify a primary PostgreSQL instance.", 17)
+	findConfigOrFail(Replica, "Must specify a replica PostgreSQL instance for failover.", 18)
```

Comment:
Refactoring to a function seems good. But, I'd prefer that we not use arbitrary exit codes for this kind of failure. I believe that the original implementation of `os.Exit(1)` was sufficient for this purpose.